src/node_api.cc

```diff
@@ -787,11 +756,8 @@ napi_env GetEnv(v8::Local<v8::Context> context) {
   // because we need to stop hard if either of them is empty.
   //
   // Re https://github.com/nodejs/node/pull/14217#discussion_r128775149
-  auto key = v8::Private::ForApi(isolate,
-      v8::String::NewFromOneByte(isolate,
-          reinterpret_cast<const uint8_t*>("N-API Environment"),
-          v8::NewStringType::kInternalized).ToLocalChecked());
-  auto value = global->GetPrivate(context, key).ToLocalChecked();
+  auto value = global->GetPrivate(context, NAPI_PRIVATE_KEY(isolate, env))
```

Comment:
Since you have a reference to the `v8::Context`, it would be more efficient to call `Environment::GetCurrent(context)` once and then call `env->napi_env()` directly.

Applies to a few more spots.  I won't point them out individually.